src/content/services/monetizationTagManager.ts

```diff
@@ -496,47 +506,30 @@ export class MonetizationTagManager extends EventEmitter {
 
     if (tag.hasAttribute('disabled')) return null
 
-    const isValidWalletAddressUrlFormat = this.checkWalletAddressUrlFormat(tag)
-
-    return isValidWalletAddressUrlFormat
-      ? await this.checkWalletAddressUrlCall(tag)
-      : null
-  }
-
-  // Check wallet address url for valid format
-  private checkWalletAddressUrlFormat(tag: MonetizationTag) {
-    const walletAddressUrl = tag.href.trim()
-
-    try {
-      checkWalletAddressUrlFormat(walletAddressUrl, true)
-
-      return true
-    } catch (err) {
-      this.logger.error(err)
-
-      const event = new Event('error')
-      tag.dispatchEvent(event)
+    const walletAddressInfo = await this.validateWalletAddress(tag)
 
-      return false
-    }
+    return walletAddressInfo
   }
 
-  // Check wallet address url by fetching the wallet address details
-  private async checkWalletAddressUrlCall(
+  private async validateWalletAddress(
     tag: MonetizationTag
   ): Promise<WalletAddress | null> {
     const walletAddressUrl = tag.href.trim()
-
     try {
+      checkWalletAddressUrlFormat(walletAddressUrl)
       const response = await checkWalletAddressUrlCall({ walletAddressUrl })
 
-      return response.success ? response.payload : null
-    } catch (err) {
-      this.logger.error(err)
-
-      const event = new Event('error')
-      tag.dispatchEvent(event)
+      if (response.success === false) {
+        throw new Error(
+          `Could not retrieve wallet address information for ${JSON.stringify(walletAddressUrl)}.`
+        )
+      }
 
+      this.dispatchLoadEvent(tag)
+      return response.payload
+    } catch (e) {
+      this.logger.error(e)
+      this.dispatchErrorEvent(tag)
```

Comment:
Should include error message in this (can dispatch `new ErrorEvent('error', {message})`), instead of logging from content script (optional instead, ok to show error twice I think).

Reply:
Looks like ErrorEvent can't be transmitted across content script boundary. Using CustomElement like we do for `monetization` event should work.

```diff
diff --git a/src/content/polyfill.ts b/src/content/polyfill.ts
index acd8c65..7c03428 100644
--- a/src/content/polyfill.ts
+++ b/src/content/polyfill.ts
@@ -122,4 +122,19 @@ import type { MonetizationEventPayload } from '@/shared/messages'
     },
     { capture: true }
   )
+
+  window.addEventListener(
+    '__wm_ext_error',
+    (event: CustomEvent<{ message: string; error: Error }>) => {
+      if (!event.target) return
+      if (!(event.target instanceof HTMLElement)) return
+      event.target.dispatchEvent(
+        new ErrorEvent('error', {
+          message: event.detail.message,
+          error: event.detail.error
+        })
+      )
+    },
+    { capture: true }
+  )
 })()
diff --git a/src/content/services/monetizationTagManager.ts b/src/content/services/monetizationTagManager.ts
index 8105c50..b5e8a8c 100644
--- a/src/content/services/monetizationTagManager.ts
+++ b/src/content/services/monetizationTagManager.ts
@@ -69,8 +69,13 @@ export class MonetizationTagManager extends EventEmitter {
     tag.dispatchEvent(new Event('load'))
   }
 
-  private dispatchErrorEvent(tag: MonetizationTag) {
-    tag.dispatchEvent(new Event('error'))
+  private dispatchErrorEvent(tag: MonetizationTag, error: Error) {
+    tag.dispatchEvent(
+      new CustomEvent('__wm_ext_error', {
+        detail: mozClone({ message: error.message, error }, this.document),
+        bubbles: true
+      })
+    )
   }
 
   dispatchMonetizationEvent({ requestId, detail }: MonetizationEventPayload) {
@@ -529,7 +534,7 @@ export class MonetizationTagManager extends EventEmitter {
       return response.payload
     } catch (e) {
       this.logger.error(e)
-      this.dispatchErrorEvent(tag)
+      this.dispatchErrorEvent(tag, e as Error)
       return null
     }
   }
```